src/object/cli_obj.c

```diff
@@ -1738,16 +1738,20 @@ obj_ec_parity_alive(daos_handle_t oh, uint64_t dkey_hash, uint32_t *shard)
 	oca = obj_get_oca(obj);
 	p_shard = obj_ec_parity_start(obj, dkey_hash);
 	for (i = 0; i < obj_ec_parity_tgt_nr(oca); i++, p_shard++) {
-		*shard = p_shard % obj_get_grp_size(obj) + grp_idx * obj_get_grp_size(obj);
-		D_DEBUG(DB_TRACE, "shard %u %d/%d/%d/%d/%d\n", *shard,
-			obj->cob_shards->do_shards[*shard].do_rebuilding,
-			obj->cob_shards->do_shards[*shard].do_reintegrating,
-			obj->cob_shards->do_shards[*shard].do_target_id,
-			obj->cob_shards->do_shards[*shard].do_shard,
-			obj->cob_shards->do_shards[*shard].do_shard_idx);
-		if (!obj_shard_is_invalid(obj, *shard, DAOS_OBJ_RPC_FETCH) &&
-		    !obj->cob_shards->do_shards[*shard].do_reintegrating)
+		uint32_t shard_idx = p_shard % obj_get_grp_size(obj) +
+				     grp_idx * obj_get_grp_size(obj);
+		D_DEBUG(DB_TRACE, "shard %u %d/%d/%d/%d/%d\n", shard_idx,
+			obj->cob_shards->do_shards[shard_idx].do_rebuilding,
+			obj->cob_shards->do_shards[shard_idx].do_reintegrating,
+			obj->cob_shards->do_shards[shard_idx].do_target_id,
+			obj->cob_shards->do_shards[shard_idx].do_shard,
+			obj->cob_shards->do_shards[shard_idx].do_shard_idx);
+		if (!obj_shard_is_invalid(obj, shard_idx, DAOS_OBJ_RPC_FETCH) &&
+		    !obj->cob_shards->do_shards[shard_idx].do_reintegrating) {
+			*shard = p_shard % daos_oclass_grp_size(&obj->cob_oca) +
```

Comment:
"p_shard" is based on the object class, then "p_shard % obj_get_grp_size(obj)" will be always within the normal shards range instead of the extended ones, right?

Reply:
> "p_shard" is based on the object class, then "p_shard % obj_get_grp_size(obj)" will be always within the normal shards range instead of the extended ones, right?

Yes, you are right, this will not touch extend shard, since only list function(from migration) will use it. Btw: it probably use obj_oclass_grp_size() might be better for reading,  but p_shard is parity offset within 1 group (0 - tgt_nr), so it is really no difference to use obj_get_grp_size() or oclass_grp_size() here.